### PR TITLE
Create `airdrop.claims` with 38 airdrops

### PR DESCRIPTION
**Pull request by hildobby**
Brief comments on the purpose of your changes:

Here I created a new sector `airdrop` since without standardisation, things are starting to get messy in spellbook. The goal is to have 3 main tables at the end (`airdrop.claims`, `airdrop.eligible`, `airdrop.total_stats`):

- `airdrop.claims` is what this PR creates, its onchain claims of airdrops
- `airdrop.eligible` would have the eligible address lists and since the breakdown is different per airdrop, this would only contain the address and total its eligible for with the breakdown still in the sub models.
- `airdrop.total_stats` would have 1 line per airdrop and consist of total claimable, % claimed, etc

All those depend on their blockchain dependant sub-abstractions (ie `airdrop.claims` depends on `airdrop_ethereum.claims` and `airdrop_optimism.claims`) in this PR. And those depend on the protocol specific tables (ie `looksrare_ethereum.airdrop_claims`).

This PR creates `airdrop.claims` with those airdrops included so far:
- Ampleforth (FORTH)
- ApeCoin (APE)
- BendDAO (BEND)
- Blur (BLUR airdrop 1)
- Cow Protocol (COW)
- DappRadar (RADAR)
- dYdX (DYDX)
- ENS (ENS)
- Forta Network (FORT)
- Galxe (GAL)
- Gas DAO (GAS)
- Gitcoin (GTC)
- Hop Protocol (HOP)
- LooksRare (LOOKS)
- 1inch (1INCH)
- Paladin (PAL)
- Paraswap (PSP)
- Ribbon (RBN)
- Safe (SAFE)
- Sudoswap (SUDO)
- Tornado Cash (TORN)
- Uniswap (UNI)
- X2Y2 (X2Y2)
- Optimism (OP airdrop 1)
- Velodrome (VELO)

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Since this is a new sector, maybe @augustog can take a look.

File: models/op/optimism/op_optimism_airdrop_1_claims.sql
Comment (MSilb7) [inline models/op/optimism/op_optimism_airdrop_1_claims.sql:23]: do we need to cast any of these as double for dunesql future-proofing? Many had decimal amounts (i.e. 778.79)

File: models/ampleforth/ethereum/ampleforth_ethereum_airdrop_claims.sql
Comment (jeff-dude) [inline models/ampleforth/ethereum/ampleforth_ethereum_airdrop_claims.sql:38]: i assume with this filter, that means the airdrop claim is over? in that case, i'd suggest we remove logic for incremental and make it a table. we'd also want to apply a 'static' tag in the config block of the model, that way our prod job doesn't run it each time as there is no new data. please look to apply this logic throughout any spell in this PR that fits that criteria

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: dbt_project.yml
Comment (jeff-dude) [inline dbt_project.yml:826]: i think for blockchain-level airdrops, we can stick to just one level in the hierarchy, rather than repeat it twice -- wdyt?

Comment (jeff-dude) [reply dbt_project.yml:826]: if changed here, of course all the models in the directory it relates to need changed as well

Comment (hildobby) [reply dbt_project.yml:826]: What do you mean exactly? What is repeated twice and where?

Comment (jeff-dude) [reply dbt_project.yml:826]: arbitrum directory within the arbitrum directory -- i.e. `models/arbitrum/arbitrum/`. i think we can just stop at `models/arbitrum/` and do it all within there and simplify naming standards on files names, directories, refs(), etc

**Comment by hildobby** [comment]
Blocked by #3058

File: dbt_project.yml
Comment (hildobby) [reply dbt_project.yml:826]: Yeah I get how than can be confusing but imo its the most futureproof method, I bet we will eventually see a blockchain airdrop its native token on a chain other than its own/multiple chains!

Also, Optimism has `op_optimism` for their token which doesn't follow the dune structure of `protocol_chain`, I was thinking that it should be renamed but it could be renamed to either `optimism_optimism` or `optimism`, imo the most futureproof is the most detailed (`optimism_optimism`), but maybe @MSilb7 & @chuxinh have some thoughts here?

**Comment by hildobby** [comment]
Not sure why it times out at 3h although I extended to 6, any ideas @jeff-dude?

Either way, I have some more airdrops to add but I want to hold off on them until this gets merged and submit them in another PR, there already are more than 40 airdrops and I included one for every chain (other than phantom) so that it's easy to work with for follow up PRs :)

File: models/thales/ethereum/thales_ethereum_airdrop_claims.sql
Comment (jeff-dude) [inline models/thales/ethereum/thales_ethereum_airdrop_claims.sql:49]: this join appears to be lacking join conditions, therefore performing joins which explode record count, i.e. why it's timing out

File: models/value_defi/ethereum/value_defi_ethereum_airdrop_claims.sql
Comment (jeff-dude) [inline models/value_defi/ethereum/value_defi_ethereum_airdrop_claims.sql:49]: same join issues here

**Review by jeff-dude — CHANGES_REQUESTED** [review]
two spells appear to hold up the CI test, commented on join conditions of both below

**Comment by jeff-dude** [comment]
can we also reconsider the approach to build this new sector spell? it's difficult to get a good review on the foundation of the spell when there are so many added within.
throwing out an idea:
- keep this feature branch on your forked repo
- add new feature branch, add one or two base level spells for airdrops as a starting point
- add the `airdrop` level schema on top
- review/merge that as a start
- come back and add more airdrops in follow up PR

**Comment by hildobby** [comment]
> can we also reconsider the approach to build this new sector spell? it's difficult to get a good review on the foundation of the spell when there are so many added within. throwing out an idea:
> 
> * keep this feature branch on your forked repo
> * add new feature branch, add one or two base level spells for airdrops as a starting point
> * add the `airdrop` level schema on top
> * review/merge that as a start
> * come back and add more airdrops in follow up PR

Alright @jeff-dude, just created #3147 that contains the core of the airdrop sector including one airdrop per chain for now, will mark this one as blocked until #3147 is merged

**Comment by jeff-dude** [comment]
sorry, more merge conflicts to resolve 😅 also -- looks like some models are maybe missing in schema file, hence the error on pre-commit hooks test?

**Comment by jeff-dude** [comment]
need to resolve some issues on why the ci test is timing out, then get a successful run as expected, then we can proceed 🙏

**Comment by jeff-dude** [comment]
looks like it timed out on dunesql check, that should be fine. i'll let it run this last time on latest merge and then we should be good

**Review by jeff-dude — APPROVED** [review]
looks like we're finally all good 🤞

**Pull request merged**
